lib/spawn/utils/nats.ex

```diff
@@ -0,0 +1,55 @@
+defmodule Spawn.Utils.Nats do
+  @moduledoc false
+
+  alias alias Eigr.Functions.Protocol.InvocationRequest
+
+  import Spawn.Utils.Common, only: [to_existing_atom_or_new: 1]
+
+  def connection_name(), do: to_existing_atom_or_new("spawn.internal.nats")
+
+  def get_internal_nats_connection(config) do
+    raw_hosts = config.internal_nats_hosts
+    hosts_conn_map = get_nats_hosts(raw_hosts)
+
+    # TODO: Get other parameters here to build complex connections
+
+    hosts_conn_map
+  end
+
+  def get_nats_connection(config) do
+    raw_hosts = config.pubsub_adapter_nats_hosts
+    hosts_conn_map = get_nats_hosts(raw_hosts)
+
+    # TODO: Get other parameters here to build complex connections
+
+    hosts_conn_map
+  end
+
+  def request(system, payload, opts \\ []) do
+    async? = Keyword.get(opts, :async, false)
+    conn = connection_name()
+    topic = "spawn.#{system}.actors.actions"
```

Comment:
:information_source: You could move this to a function in this module and reuse it in the supervisor